src/Admin/ValidationCounts.php

```diff
@@ -122,12 +123,19 @@ public function enqueue_scripts() {
 	 * screen related to `amp_validation_error` post type (which includes the `amp_validation_error` taxonomy).
 	 */
 	protected function maybe_add_preload_rest_paths() {
-		if (
+		if ( $this->is_dedicated_plugin_screen() ) {
+			$this->rest_preloader->add_preloaded_path( '/amp/v1/unreviewed-validation-counts' );
+		}
+	}
+
+	/**
+	 * Whether the current screen is pages inside the AMP Options menu.
+	 */
+	public function is_dedicated_plugin_screen() {
+		return (
 			AMP_Options_Manager::OPTION_NAME === get_admin_page_parent()
 			||
 			AMP_Validated_URL_Post_Type::POST_TYPE_SLUG === get_current_screen()->post_type
-		) {
-			$this->rest_preloader->add_preloaded_path( '/amp/v1/unreviewed-validation-counts' );
-		}
+		);
```

Comment:
For `is_needed` the conditions may need to be expanded to account for all of these screens and subscreens:

![image](https://user-images.githubusercontent.com/134745/217340220-5504e7a3-3785-47c6-98aa-c701fe77bccc.png)

Reply:
Do you want the `ValidationCounts` service to be needed only on the AMP settings menus?

Reply:
It can be on all screens when we core is shipping the expected version of React, but otherwise it can be limited to only the AMP settings pages.

Reply:
`ValidationCounts` doesn't have any dependency on React and uses `'wp-api-fetch', 'wp-dom-ready', 'wp-i18n'`(which also doesn't have any dependency on React). So I think we can keep it on all screens.

Reply:
Ah, good point. Nevertheless, we will need to _not_ register the polyfills for this since the polyfills are unconditional now.

Reply:
That is, unless we limit the service to only run on our dedicated screens. But probably simpler to just remove the polyfill registration entirely for this. Since the service is already conditional based on whether it `has_support`, there doesn't seem to be any need for polyfills anyway, right?

Reply:
Yes, we can remove polyfills for it entirely as it's conditional based on `has_support`.

Reply:
Removed the `amp_register_polyfills` hook from the `ValidationCounts` service.